manim/utils/hashing.py

```diff
@@ -0,0 +1,150 @@
+        """
+        This method is used to serialize objects to JSON format.
+        If obj is a function, then it will return a dict with two keys : 'code', for the code source, and 'nonlocals' for all nonlocalsvalues. (including nonlocals functions, that will be serialized as this is recursive.)
+        if obj is a np.darray, it converts it into a list.
+        if obj is an object with __dict__ attribute, it returns its __dict__.
+        Else, will let the JSONEncoder do the stuff, and throw an error if the type is not suitable for JSONEncoder. 
+
+        Parameters
+        ----------
+        obj : `any`
+            Arbitrary object to convert
+
+        Returns
+        -------
+        `any`
+            Python object that JSON encoder will recognize
+
+        """
+        if inspect.isfunction(obj) and not isinstance(obj, ModuleType):
+            r = inspect.getclosurevars(obj)
+            x = {**copy.copy(r.globals), **copy.copy(r.nonlocals)}
+            for i in list(x):
+                # NOTE : All module types objects are removed, because otherwise it throws ValueError: Circular reference detected if not. TODO
+                if isinstance(x[i], ModuleType):
+                    del x[i]
+            return {'code': inspect.getsource(obj),
+                    'nonlocals': x}
+        elif isinstance(obj, np.ndarray):
+            return list(obj)
+        elif hasattr(obj, "__dict__"):
+            return getattr(obj, '__dict__')
+        elif isinstance(obj, np.uint8):
+            return int(obj)
+        try:
+            return json.JSONEncoder.default(self, obj)
+        except TypeError:
+            # This is used when the user enters an unknown type in CONFIG. Rather than throwing an error, we transform
+            # it into a string "Unsupported type for hashing" so it won't affect the hash.
+            return "Unsupported type for hashing"
+
+
+def get_json(object):
+    """Recursively serialize object to JSON. Use CustomEncoder class above.
+
+    Paramaters
+    ----------
+    dict_config : :class:`dict`
+        The dict to flatten
+
+    Returns
+    -------
+    `str` 
+        the object flattened
+    """
+    return json.dumps(object, cls=CustomEncoder)
+
+
+def get_camera_dict_for_hashing(camera_object):
+    """Remove some keys from cameraobject.__dict__ that are useless for the caching functionnality and very heavy. 
+
+    Parameters
+    ----------
+    object_camera : :class:``~.Camera`
+        The camera object used in the scene
+
+    Returns
+    -------
+    `dict`
+        Camera.__dict__ but cleaned.
+    """
+    camera_object_dict = copy.copy(camera_object.__dict__)
+    # We have to clean a little bit camera_dict, as pixel_array and background are two very big numpy array.
+    # They are not essential to caching process.
+    # We also have to remove pixel_array_to_cairo_context as it conntains uses memory adress (set randomly). See l.516 get_cached_cairo_context in camera.py
```

Comment:
Contains? Uses? Contains used?
```suggestion
    # We also have to remove pixel_array_to_cairo_context as it contains used memory adress (set randomly). See l.516 get_cached_cairo_context in camera.py
```